src/sage/schemes/curves/zariski_vankampen.py

```diff
@@ -844,71 +866,74 @@ def braid_in_segment(glist, x0, x1, precision={}):
             intervals[f] = [r.interval(CIFp) for r in y0sf]
-            if not any(a.overlaps(b) for a, b in itertools.combinations(intervals[f], 2)):
+            if not any(a.overlaps(b) for a, b in
+                       itertools.combinations(intervals[f], 2)):
                 break
             precision1[f] *= 2
     strands = []
     for f in glist:
         for i in intervals[f]:
-            aux = followstrand(f, [p for p in glist if p != f], x0, x1, i.center(), precision1[f])
+            aux = followstrand(f, [p for p in glist if p != f],
+                               x0, x1, i.center(), precision1[f])
             strands.append(aux)
-    complexstrands = [[(QQ(a[0]), QQ(a[1]), QQ(a[2])) for a in b] for b in strands]
+    complexstrands = [[(QQ(a[0]), QQ(a[1]), QQ(a[2])) for a in b]
+                      for b in strands]
     centralbraid = braid_from_piecewise(complexstrands)
     initialstrands = []
     finalstrands = []
-    initialintervals = roots_interval_cached(g, X0)
-    finalintervals = roots_interval_cached(g, X1)
+    initialintervals = roots_interval_cached(g, x0)
+    finalintervals = roots_interval_cached(g, x1)
     I1 = QQbar.gen()
     for cs in complexstrands:
         ip = cs[0][1] + I1 * cs[0][2]
         fp = cs[-1][1] + I1 * cs[-1][2]
         matched = 0
         for center, interval in initialintervals.items():
             if ip in interval:
-                initialstrands.append([(0, center.real(), center.imag()), (1, cs[0][1], cs[0][2])])
+                initialstrands.append([(0, center.real(), center.imag()),
+                                       (1, cs[0][1], cs[0][2])])
                 matched += 1
         if matched != 1:
-            precision1 = {f: precision1[f] * 2 for f in glist}  # new
-            return braid_in_segment(glist, x0, x1, precision=precision1)  # new
+            precision1 = {f: precision1[f] * 2 for f in glist}
+            return braid_in_segment(glist, x0, x1, precision=precision1)
 
         matched = 0
         for center, interval in finalintervals.items():
             if fp in interval:
-                finalstrands.append([(0, cs[-1][1], cs[-1][2]), (1, center.real(), center.imag())])
+                finalstrands.append([(0, cs[-1][1], cs[-1][2]),
+                                     (1, center.real(), center.imag())])
                 matched += 1
         if matched != 1:
-            precision1 = {f: precision1[f] * 2 for f in glist}  # new
-            return braid_in_segment(glist, x0, x1, precision=precision1)  # new
+            precision1 = {f: precision1[f] * 2 for f in glist}
+            return braid_in_segment(glist, x0, x1, precision=precision1)
     initialbraid = braid_from_piecewise(initialstrands)
     finalbraid = braid_from_piecewise(finalstrands)
 
     return initialbraid * centralbraid * finalbraid
 
 
-def geometric_basis(G, E, EC0, p, dual_graph):
+def geometric_basis(G, E, EC0, p, dual_graph, vertical_regions):
```

Comment:
While this is not imported, it is part of our public documentation and this represents a backwards incompatible change. I think we should allow the previous input to work if possible.

Reply:
As far as I know `geometric_basis` is used only here and this would be part of the public documentation. The use of this *vertical* objects makes computations faster. Let me think if added as optional does not break the code (it shouldn't)

Reply:
Now it is an optional input